packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts

```diff
@@ -781,17 +1024,135 @@ export class GaussianSplattingMesh extends Mesh {
                 textureSize.y,
                 this._useRGBACovariants ? Constants.TEXTUREFORMAT_RGBA : Constants.TEXTUREFORMAT_RG
             );
-            this._centersTexture = createTextureFromData(this._splatPositions, textureSize.x, textureSize.y, Constants.TEXTUREFORMAT_RGBA);
+            this._centersTexture = createTextureFromData(this._splatPositions!, textureSize.x, textureSize.y, Constants.TEXTUREFORMAT_RGBA);
             this._colorsTexture = createTextureFromDataU8(colorArray, textureSize.x, textureSize.y, Constants.TEXTUREFORMAT_RGBA);
             this._instanciateWorker();
         }
     }
 
-    private _loadData(data: ArrayBuffer): void {
+    private *_updateDataAsync(data: ArrayBuffer): Coroutine<void> {

```

Comment:
Seems weird to use the Async suffix here, since the function itself is not async (it depends on how the coroutine is run). Maybe just name it _updateDataCoroutine?

Reply:
I get TS errors with yield not found.